### DISCUSSION TITLE
Inserting a page before the table of content in a Quarto book (`.pdf` format)

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#11386] [Q&A] [answered]
### Description

I've a Quarto book project. I'd like to add a page before the table of content. Tweaking template partials seems to be the solution (#7389).

Instead of copying the content of the `.qmd` file into the `.tex` file and styling it manually with LaTeX, I'm wondering if there's a way to reference the `.qmd` file directly in the `.tex` file.

For example, considering the following `firstpage.qmd` that I'd like to add at the end of `before-body.tex`:

```qmd
# First page {.unnumbered .unlisted}

Qui cupidatat minim tempor occaecat labore esse magna minim. Amet labore cupidatat eiusmod sunt. Labore aute proident cupidatat esse laborum tempor officia elit cupidatat. Aute nisi reprehenderit cillum ex laborum duis reprehenderit occaecat laborum eiusmod sunt ea et.

Veniam consequat elit dolore aliqua. Culpa mollit exercitation nulla aute in culpa. Occaecat anim Lorem nisi laboris elit velit aliqua deserunt nisi consequat.

- **a:** Velit voluptate elit ad anim cillum.
- **b:** Nisi magna fugiat veniam enim quis enim.
```

Is it possible to do it without rewriting everything in the `.tex` file? I'm rendering the book in `.pdf`.

`before-body.tex`:
```tex
$if(has-frontmatter)$
\frontmatter
$endif$
$if(title)$
$if(beamer)$
\frame{\titlepage}
$else$
\maketitle
$endif$
$if(abstract)$
\begin{abstract}
$abstract$
\end{abstract}
$endif$
$endif$
% add a command/hook to render firstpage.qmd here
```

I've uploaded a dummy repo [here](https://github.com/arnaudgallou/partials) if need be.

**Comment by mcanouil** [comment]
No, you can't have markdown inside LaTeX. LaTeX does not know how to read it. It needs to be parsed first.

So, write LaTeX into a file, include that file using Pandoc templating system inside the partial of your choice.

**Reply by arnaudgallou** [reply]
Of course I can't have the markdown directly in the `.tex` file. But since the markdown will eventually generate LaTeX, isn't it possible to have a hook to tell Quarto where a particular page should be?

Something similar to the `body` placeholder in the following example:

```tex
\documentclass{scrartcl}
\begin{document}
$body$
\end{document}
```

**Reply by mcanouil** [reply]
Yes, you could do that likely with a Lua filter.

Basically the other way around of the following: https://github.com/pandoc-ext/abstract-section

**Reply by arnaudgallou** [reply]
Alright thanks. I was hoping for a simpler solution.

When you say:
>  include that file using Pandoc templating system

Do you mean something like `$myfile.tex()$` and add that file to the `template-partials` option in `_quarto.yml`?

**Reply by mcanouil** [reply]
As it's not an "official" partial, I'm not sure you can control it `template-partials` (worth trying though).
I was mostly saying to use Pandoc template syntax which indeed is: `$myfile.tex()$`

**Reply by arnaudgallou** [reply]
In fact, I could only make it work by adding the file to `template-partials`. Otherwise I get the following error:

```
Could not find data file templates/myfile.tex
```

Since that's not documented I wasn't sure this is the right way to do it.

Also, one of the issue I noticed is that the content in `myfile.tex` doesn't inherit all the layout options defined in `_quarto.yml`. E.g., the line stretch on that particular page seems to use the default value. Not sure if this is normal though.

**Reply by mcanouil** [reply]
> Also, one of the issue I noticed is that the content in myfile.tex doesn't inherit all the layout options defined in _quarto.yml. E.g., the line stretch on that particular page seems to use the default value. Not sure if this is normal though.

Are you referring to the discussion you opened?
- https://github.com/quarto-dev/quarto-cli/discussions/11391

**Reply by arnaudgallou** [reply]
I don't think it's related that's why I mentioned it here. The problem in #11391 is more general since it affects all pages and doesn't reset options initially set such as the font or line stretch.

What I'm talking about here is specific to the content in `myfile.tex` and seems to only affects specific options like the line stretch. Custom fonts are correctly inherited.

Not sure if that's actually a bug or if I'm supposed to set the line stretch manually in `myfile.tex`. That's why I didn't open a discussion about it.

**Comment by cderv** [comment, marked as answer]
FWIW someone in the past manage to change placement for pdf output by suppressing toc handling by Quarto / pandoc with `toc: false` and inserting raw latex directly in body where to put it

- See https://github.com/quarto-dev/quarto-cli/discussions/9839#discussioncomment-9629083

Anything written in the .qmd will be in `$body$` so that is one way to achieve it. Otherwise, playing with metadata variable and pandoc's template is the way for this. But the first part before toc will need to be passed a pandoc variable (directly from YAML), or using similar tricks as [pandoc-ext/abstract-section](https://github.com/pandoc-ext/abstract-section) indeed.

**Reply by arnaudgallou** [reply]
Using `toc: false` and placing `\tableofcontents` manually in the book is an even simpler solution than tweaking template partials and doesn't have the formatting/layout issues mentioned in the posts above.
